catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoListener.java

```diff
@@ -87,24 +87,15 @@ public void onDigitalMessageReceived(DigitalMessage message) {
 
 		switch (message.getPort()) {
 			case ArduinoImpl.PORT_DIGITAL_0:
-				portValue [2] = (message.getValue() & 0x4) == 0 ? 0 : 1;
-				portValue [3] = (message.getValue() & 0x8) == 0 ? 0 : 1;
-				portValue [4] = (message.getValue() & 0x16) == 0 ? 0 : 1;
-				portValue [5] = (message.getValue() & 0x32) == 0 ? 0 : 1;
-				portValue [6] = (message.getValue() & 0x64) == 0 ? 0 : 1;
+				portValue[0] = message.getValue();
 				break;
 			case ArduinoImpl.PORT_DIGITAL_1:
-				portValue [8] = (message.getValue() & 0x1) == 0 ? 0 : 1;
-				portValue [9] = (message.getValue() & 0x2) == 0 ? 0 : 1;
-				portValue [10] = (message.getValue() & 0x4) == 0 ? 0 : 1;
-				portValue [11] = (message.getValue() & 0x8) == 0 ? 0 : 1;
-				portValue [12] = (message.getValue() & 0x16) == 0 ? 0 : 1;
-				portValue [13] = (message.getValue() & 0x32) == 0 ? 0 : 1;
+				portValue[1] = message.getValue();
```

Comment:
`portValue[0]` is aka `portValue[PORT_DIGITAL_0]` and `portValue[1]` is aka `portValue[PORT_DIGITAL_1]`.
So you could remove the switch statement and use `portValue[message.getPort()]` instead.